testing/auth.ts

```diff
@@ -284,6 +292,44 @@ class CredentialHandler {
     });
   }
 
+  handleOAuth2ClientCredentials() {
+    assertCondition(this._authDef.type === AuthenticationType.OAuth2ClientCredentials);
+    const existingCredentials = this.checkForExistingCredential() as OAuth2ClientCredentials | undefined;
+    const {clientId, clientSecret} = this._promptOAuth2ClientIdAndSecret(existingCredentials);
+    const credentials: OAuth2ClientCredentials = {
+      clientId,
+      clientSecret,
+      accessToken: existingCredentials?.accessToken,
+      expires: existingCredentials?.expires,
+      scopes: existingCredentials?.scopes,
+    };
+    this.storeCredential(credentials);
+
+    const manifestScopes = this._authDef.scopes || [];
+    const requestedScopes =
+        this._extraOAuthScopes.length > 0 ? [...manifestScopes, ...this._extraOAuthScopes] : manifestScopes;
+
+    performOAuthClientCredentialsServerFlow({
+      clientId,
+      clientSecret,
+      scopes: requestedScopes,
+      authDef: this._authDef,
+    }).then(({accessToken, expires}) => {
```

Comment:
Hmm let's not use then/catch. If you want to kick this off in this non-async method, would just make a new private method that does it, can just `void` that method when you call it to orphan the promise. Would leave a TODO that we should go back make input handling async if we're going to do stuff like this inside of it.

Either way this feels a little brittle though because I think the script can exit while this orphaned promise is still running, and so your CLI output might look weird. Not sure if there's a way to wait for all promise to settle before exciting the CLI main.